Sources/PackageCollections/PackageCollections.swift

```diff
@@ -0,0 +1,296 @@
+
+        self.storage.collectionsProfiles.listSources(in: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let sources):
+                if sources.isEmpty {
+                    return callback(.success([]))
+                }
+                let lock = Lock()
+                var refreshResults = [Result<PackageCollectionsModel.Collection, Error>]()
+                sources.forEach { source in
+                    self.refreshPackageGroupFromSource(source: source, profile: profile) { refreshResult in
+                        lock.withLock { refreshResults.append(refreshResult) }
+                        if refreshResults.count == (lock.withLock { sources.count }) {
+                            let errors = refreshResults.compactMap { $0.failure }
+                            callback(errors.isEmpty ? .success(sources) : .failure(MultipleErrors(errors)))
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    func addCollection(_ source: PackageCollectionsModel.CollectionSource,
+                       order: Int? = nil,
+                       to profile: PackageCollectionsModel.Profile? = nil,
+                       callback: @escaping (Result<PackageCollectionsModel.Collection, Error>) -> Void) {
+        let profile = profile ?? .default
+
+        if let errors = source.validate() {
+            return callback(.failure(MultipleErrors(errors)))
+        }
+
+        self.storage.collectionsProfiles.add(source: source, order: order, to: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success:
+                self.refreshPackageGroupFromSource(source: source, order: order, profile: profile, callback: callback)
+            }
+        }
+    }
+
+    func removeCollection(_ source: PackageCollectionsModel.CollectionSource,
+                          from profile: PackageCollectionsModel.Profile? = nil,
+                          callback: @escaping (Result<Void, Error>) -> Void) {
+        let profile = profile ?? .default
+
+        self.storage.collectionsProfiles.remove(source: source, from: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success:
+                self.storage.collectionsProfiles.exists(source: source, in: nil) { result in
+                    switch result {
+                    case .failure(let error):
+                        callback(.failure(error))
+                    case .success(let exists):
+                        if exists {
+                            callback(.success(()))
+                        } else {
+                            self.storage.collections.remove(identifier: .init(from: source), callback: callback)
```

Comment:
Let's add a code comment to explain that we are deleting a package collection from storage when it's no longer a part of any profile.